src/words/ast/INodeAssign.java

```diff
@@ -8,9 +10,36 @@ public INodeAssign(Object... children) {
 		super(children);
 	}
 
+	/**
+	 * Always returns null. Throws WordsRuntimeException for called methods, including WordsObjectNotFoundException.
+	 * Side effect of assigning a property to an object.
+	 */
 	@Override
 	public ASTValue eval(WordsEnvironment environment) throws WordsRuntimeException {
-		// TODO
-		throw new AssertionError("Not yet implemented");
+		WordsObject obj = children.get(0).eval(environment).objValue;
```

Comment:
Ah, I see what you're saying. You're right I depended on the grammar enforcing that. Since at least one reference must be in the list for that production, there's either an object or an exception will be thrown. I'll throw the AssertionError. Am I correct in understanding our use of the AssertionError as something that should never happen under the current way we have things set up?